internal/cli/project_inspect.go

```diff
@@ -0,0 +1,248 @@
+)
+
+type ProjectInspectCommand struct {
+	*baseCommand
+
+	flagJson bool
+}
+
+func (c *ProjectInspectCommand) Run(args []string) int {
+	flagSet := c.Flags()
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithConfig(true), // optional config loading
+	); err != nil {
+		return 1
+	}
+
+	cmdArgs := flagSet.Args()
+	var projectTarget string
+	if len(cmdArgs) > 1 {
+		c.ui.Output("No more than 1 argument required.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	} else if len(cmdArgs) == 0 {
+		// If we're in a project dir, load the name. Otherwise we'll
+		// try the arg passed in
+		if c.project.Ref() != nil {
+			projectTarget = c.project.Ref().Project
+		} else {
+			c.ui.Output("Project argument required, and not in a project directory..\n\n"+
+				c.Help(), terminal.WithErrorStyle())
+			return 1
+		}
+	} else if len(cmdArgs) == 1 {
+		// project requested
+		projectTarget = cmdArgs[0]
+	}
+
+	err := c.FormatProject(projectTarget)
+	if err != nil {
+		c.ui.Output("Failed to format project: %s"+
+			clierrors.Humanize(err),
+			terminal.WithErrorStyle())
+		return 1
+	}
+	return 0
+}
+
+func (c *ProjectInspectCommand) FormatProject(projectTarget string) error {
+	// Get our API client
+	client := c.project.Client()
+
+	resp, err := client.GetProject(c.Ctx, &pb.GetProjectRequest{
+		Project: &pb.Ref_Project{
+			Project: projectTarget,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	project := resp.Project
+	workspaces := resp.Workspaces
+
+	var appNames []string
+	for _, app := range project.Applications {
+		appNames = append(appNames, app.Name)
+	}
+
+	var workspaceNames []string
+	for _, ws := range workspaces {
+		workspaceNames = append(workspaceNames, ws.Workspace.Workspace)
+	}
+
+	var gitUrl, gitRef, gitPath string
+	dataSource := "Local" // if unset, assume local
+	if project.DataSource != nil {
+		switch ds := project.DataSource.Source.(type) {
+		case *pb.Job_DataSource_Local:
+			dataSource = "Local"
+		case *pb.Job_DataSource_Git:
+			dataSource = "Git"
+
+			gitUrl = ds.Git.Url
+			gitRef = ds.Git.Ref
+			gitPath = ds.Git.Path
+		}
+	}
+
+	var datasourcePollEnabled bool
+	var datasourcePollInterval string
+	if project.DataSourcePoll != nil {
+		datasourcePollEnabled = project.DataSourcePoll.Enabled
+		datasourcePollInterval = project.DataSourcePoll.Interval
+	}
+
+	var appPollEnabled bool
+	var appPollInterval string
+	if project.StatusReportPoll != nil {
+		appPollEnabled = project.StatusReportPoll.Enabled
+		appPollInterval = project.StatusReportPoll.Interval
+	}
+
+	fileChangeSignal := project.FileChangeSignal
+
+	if c.flagJson {
+		projectHeaders := []string{
+			"Project", "Applications", "Workspaces", "Remote Enabled", "Data Source",
+			"Git URL", "Git Ref", "Git Path", "Data Source Poll Enabled",
+			"Data Source Poll Interval", "App Status Poll Enabled", "App Status Poll Interval",
+			"File Change Signal",
+		}
```

Comment:
This is fine if we want to do this (I know we do this in other commands), but for the JSON output I've been just proto-json-encoding w/ `jsonpb` so that it shows the full API structure so as we add new fields I can easily see the values without having to modify the CLI (i.e. if they're not particularly useful fields but I still want to see them for debugging). Not sure if we want to maintain a stable JSON output format here or do something like that.

Reply:
Hm, I did notice that on the ODR list command! I like the idea of having a stable json output at least, but showing the complete picture for a project or other proto would be nice too. Maybe we could adopt the pattern for when we have json output to also include a `-json-raw` that does this? Maybe that's over complicated 🤔

Reply:
We will [eventually] have a pretty strict backwards compatibility requirement over our API and I believe it'll be easier to maintain compatibility dumping an API object than maintaining a CLI, just my opinion. APIs you _have_ to have backwards compat at a certain point of project maturity, but CLIs we've never covered in our entire portfolio.

Reply:
That's a good point, I'll switch this over to just show the raw project output instead.